### ISSUE TITLE
[sklearn#33783] [Callbacks] Add mechanism to realiably re-set `max_subtasks` on `CallbackContext`?

### ISSUE THREAD

**Issue by godofecht** [flooooooooooow/flow#379]
**Mirrored from [scikit-learn#33783](https://github.com/scikit-learn/scikit-learn/issues/33783)**

Original title: [Callbacks] Add mechanism to realiably re-set `max_subtasks` on `CallbackContext`?
Original labels: Callbacks
Created: 2026-04-17T10:05:30Z

---

> [!WARNING]
> This issue is not yet ready for a PR. If you are interested in contributing to scikit-learn, please have a look at our [contributing guidelines](https://scikit-learn.org/dev/developers/contributing.html), and in particular the sections for [new contributors](https://scikit-learn.org/dev/developers/contributing.html#new-contributors) and the ["Needs triage"](https://scikit-learn.org/dev/developers/contributing.html#issues-tagged-needs-triage) label.

Not a high priority, but as @adrinjalali hinted and we discussed yesterday in the meeting:

We may consider to add a mechanism to re-set the  `max_subtasks` on `CallbackContext`, so that progress bars can be re-scaled when new information on the maximum number of sub-tasks is available on an estimator or meta-estimator.

Currently, setting the `max_subtasks` after creating a new subcontext, we'd not be sure if this would update the `total` on a `ProgressBar`.

---

## Flow context

This issue is mirrored from scikit-learn to explore whether Flow encounters the same problem or can approach it differently. Add analysis below.

**Comment by godofecht** [comment]
Moved to [18](https://github.com/godofecht/flow-scikit/issues/18) in the flow-scikit repository. Further library-specific triage and implementation will happen there.

**Comment by godofecht** [comment]
Moved to [19](https://github.com/godofecht/flow-scikit/issues/19) in the flow-scikit repository. Further library-specific triage and implementation will happen there.
